src/mngt/ocf_mngt_cache.c

```diff
@@ -1791,36 +1834,47 @@ static void _ocf_mngt_cache_attach(ocf_cache_t cache,
 	context->cache = cache;
 	context->cfg = *cfg;
 
-	data = env_vmalloc(cfg->uuid.size);
-	if (!data) {
-		result = -OCF_ERR_NO_MEM;
-		goto err_pipeline;
+	if (cache->metadata.is_volatile) {
+		context->init_mode = ocf_init_mode_metadata_volatile;
+	} else {
+		context->init_mode = ocf_init_mode_init;
 	}
 
-	result = env_memcpy(data, cfg->uuid.size, cfg->uuid.data,
-			cfg->uuid.size);
-	if (result)
-		goto err_uuid;
+	OCF_PL_NEXT_RET(pipeline);
+}
 
-	context->cfg.uuid.data = data;
+static void _ocf_mngt_cache_load(ocf_cache_t cache,
```

Comment:
Why don't just move contents of this function to `ocf_mngt_cache_load()`? And the same about attach. Those functions are now really small and this additional indirection level brings no value, it only makes code more difficult to analyze.

And actually as we have separate load/attach paths, we could also get rid of `_ocf_mngt_cache_attach_complete()`/`_ocf_mngt_cache_load_complete()` and just move them to pipeline finish. The less callback/priv pairs the better for code readability.

Reply:
this step is also used in ocf_mngt_cache_attach, it is moved to pipeline for the purpose of reusing in both attach and load